src/visitor.rs

```diff
@@ -437,7 +441,8 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
                 );
                 self.push_rewrite(item.span, rewrite);
             }
-        }
+        };
+        self.skip_macro_names.borrow_mut().clear();
```

Comment:
Instead of clearing, we should restore the old `skip_macro_names` so that we don't lose the original names in `skip_macro_names` when `rustfmt::skip::macros` are nested:

```rust
// in lib.rs
#![rustfmt::skip::macros(some_macro)]

mod foo;

// in foo.rs
#[rustfmt::skip::macros(another_macro)]
fn foo() {
    another_macro!();
}

fn bar() {
    some_macro!(); // This should be skipped.
}